src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -1793,10 +1795,15 @@ open_common(int (*real_open)(const char *pathname, int oflags, ...), const char
 		rc = dfs_lookup_rel(dfs_mt->dfs, parent, item_name, oflags, &dfs_obj, &mode_query,
 				    NULL);
 	}
-
 	if (rc)
 		D_GOTO(out_error, rc);
 
+	if ((oflags & O_TRUNC) && S_ISREG(mode_query)) {
+		rc = dfs_punch(dfs_mt->dfs, dfs_obj, 0, DFS_MAX_FSIZE);
```

Comment:
you could just call dfs_open with O_TRUNC

Reply:
ah that's a bug then. yes lookup should support O_TRUNC since it's basically an open.